crates/pop-cli/src/common/wallet.rs

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::wallet_integration::{FrontendFromString, TransactionData, WalletIntegrationManager};
+use cliclack::log;
+
+pub async fn wait_for_signature(call_data: Vec<u8>, url: String) -> anyhow::Result<Option<String>> {
+	let ui = FrontendFromString::new(include_str!("../assets/index.html").to_string());
+
+	let transaction_data = TransactionData::new(url, call_data);
+	// starts server
+	let mut wallet = WalletIntegrationManager::new(ui, transaction_data);
+	log::step(format!("Wallet signing portal started at http://{}", wallet.rpc_url))?;
+
+	log::step("Waiting for signature... Press Ctrl+C to terminate early.")?;
```

Comment:
Can we change this to a spinner please? It should provide better visual feedback, but then also disappear once the signed payload is received, cleaning up the terminal a little. Otherwise there are various elements which appear as noise once done.

Reply:
https://github.com/r0gue-io/pop-cli/pull/371/commits/6035fd86b7ca679127b5cd34765a51dca6a66f13#diff-f1b7ff543f883f3dc2a2c8b3fc1c93e144e63e718655d11893011bc254f32e47R28